modules/ding_gatewayf/ding_gatewayf.module

```diff
@@ -241,6 +246,70 @@ function ding_gatewayf_callback() {
 }
 
 /**
+ * Do a check on the checksum given from gatewayf.
+ *
+ * @param array $attributes
+ */
+function ding_gatewayf_security_check($attributes) {
+  //Check attributes.
+  if (empty($attributes['schacPersonalUniqueID']) || empty($attributes['eduPersonTargetedID'])) {
+    // We expect alway to get these two attributes.
+    watchdog('ding_gatewayf', 'Neccesary attributes missing', array(), WATCHDOG_ERROR);
+    drupal_set_message('Sorry login failed - contact site administrator', 'warning');
+    ding_gatewayf_user_logout();
+  }
```

Comment:
you're right - there is a double check on one of the attributes (schacPersonalUniqueID). I will fix it and move the call to securitycheck